toolchain/check/handle_impl.cpp

```diff
@@ -176,18 +176,21 @@ static auto ExtendImpl(Context& context, Parse::NodeId extend_node,
 
   if (!context.types().Is<SemIR::FacetType>(constraint_id)) {
     context.TODO(node_id, "extending non-facet-type constraint");
-    parent_scope.has_error = true;
+    parent_scope.set_has_error();
     return;
   }
-  parent_scope.has_error |= !context.TryToDefineType(constraint_id, [&] {
-    CARBON_DIAGNOSTIC(ExtendUndefinedInterface, Error,
-                      "`extend impl` requires a definition for facet type {0}",
-                      InstIdAsType);
-    return context.emitter().Build(node_id, ExtendUndefinedInterface,
-                                   constraint_inst_id);
-  });
-
-  parent_scope.extended_scopes.push_back(constraint_inst_id);
+  if (!parent_scope.has_error() && !context.TryToDefineType(constraint_id, [&] {
```

Comment:
```suggestion
  if (!context.TryToDefineType(constraint_id, [&] {
```

I think this is actually a behavior change here, the prior `|=` is not short-circuited.

Reply:
Oops!
Done.